### ISSUE TITLE
window.location.replace not working inside cy.window()

### ISSUE THREAD

**Issue by davidhennemann** [cypress-io/cypress#5376]
### Current behavior:

After visiting a site through `cy.visit` and accessing the window object with: 
`cy.window().then(win => {...})` we can not change the hash of the site with `win.location.replace('#bar')`. Using `win.location.hash = '#bar'` instead works flawlessly. 

Calling `win.location.replace` redirects the iFrame of the tested page to a URL with a `__` prefix, the `__` could come from the main document window? (where the cypress runner lives). 

1. the runner loads `index.html`
![image](https://user-images.githubusercontent.com/6185006/66873905-7dea9280-ef5e-11e9-93b5-309bdcca653b.png)
2. calling `win.location.replace("#bar");` within `cy.window().then(win => {...})` should lead to URL `http://localhost:52402/index.html#bar`, but instead leads to:
![image](https://user-images.githubusercontent.com/6185006/66873985-b5593f00-ef5e-11e9-86ef-2051525dc8d5.png)

This is an issue for us, because we currently try to test a website, which heavily uses `win.location.replace` and if the cypress runner opens this site in the iFrame the window-object of the website somehow gets altered through cypress and the `win.location.replace` leads to the redirection error described above. **The error occurs irrespective of whether we call the redirect function from the `cy.window()`-callback or directly from within the website context.** Any idea why this is happening?

### Desired behavior:
`window.location.replace` should just change the hash and not crash the page

### Steps to reproduce: 
I created a small repo where the error is reproduced in a minimal setup. 

https://github.com/davidhennemann/cypressBugDemo

Basically just open any website and afterwards execute:
```js
it("should work", () => {
  cy.visit("https://cypress.io");

  cy.wait(2000);
  cy.window().then(win => {
    // this works
    win.location.hash = "#foo";
    // this will lead to a cypress crash
    win.location.replace("#bar");
  });
});
```

)
 ⚠️   chromeWebSecurity needs to be disable o/w an error about the redirect is thrown...

### Versions
cypress version: 3.4.1
OS: macOS

**Comment by DoubLL** [comment]
I am having the same issue on a page which is supposed to redirect the user with a script using window.location.replace().

> window.location.replace(example.html);
redirects to "localhost/__/example.html"
and not, as I would have expected, to "localhost/example.html"
which results in a failed test.

**Comment by jennifer-shehane** [comment]
I can confirm this is happening. Closing as duplicate of https://github.com/cypress-io/cypress/issues/3994 which details this issue with window.location.replace.

**Comment by matias2018** [comment]
I have the same issue, this time with window.location.assign('/mySecondPage.html');
